public/components/markdown_with_blink_cursor.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiMarkdownFormat, EuiMarkdownFormatProps } from '@elastic/eui';
+import React from 'react';
+import { useRef } from 'react';
+import { useEffect } from 'react';
+
+function getLastTextNode(node: HTMLElement | Node): Node | null | undefined {
+  if (node.nodeType === node.TEXT_NODE) {
+    return node;
+  }
+  const children = node.childNodes;
+  for (let i = children.length - 1; i >= 0; i--) {
+    const child = children[i];
+    const result = getLastTextNode(child);
+    if (result) {
+      return result;
+    }
+    return null;
+  }
+}
```

Comment:
How about
```
function isTextNode(node: Node): node is Text {
  return node.nodeType === Node.TEXT_NODE;
}

function getLastTextNode(node: HTMLElement | Node): Node | null | undefined {
  if (isTextNode(node)) {
    return node;
  }
  const children = node.childNodes;
  for (let i = children.length - 1; i >= 0; i--) {
    const child = children[i];
    if (isTextNode(child) && (child as Text).data === '\n') {
      continue;
    }
    const result = getLastTextNode(child);
    if (result) {
      return result;
    }
    return null;
  }
}
```
So the blink mark doesn't append to the next line when the current node is `li` element.

Reply:
We can have a follow up PR for this enhancement. This PR is a little bit large.

Reply:
Refactor the node and address the issue, please take a look.